editor/editor_help.cpp

```diff
@@ -74,6 +77,49 @@ class DocCache : public Resource {
 	void set_classes(const Array &p_classes) { classes = p_classes; }
 };
 
+static bool _attempt_doc_load(const String &p_class) {
+	// Docgen always happens in the outer-most class: it also generates docs for inner classes.
+	String outer_class = p_class.get_slice(".", 0);
+	if (!ScriptServer::is_global_class(outer_class)) {
+		return false;
+	}
+
+	// ResourceLoader is used in order to have a script-agnostic way to load scripts.
+	// This forces GDScript to compile the code, which is unnecessary for docgen, but it's a good compromise right now.
```

Comment:
This makes me think that perhaps the ScriptServer should be aware of documentation and just as the script language to provide documentation data. This would allow not only GDScript and other languages to optimize the doc gen process.

Of course this is just something to discuss later, no need to worry about it for this PR. I agree that loading the resource is a good compromise for now.